### PR TITLE
Add onnx export of models with a multiple choice classification head

### PR DESCRIPTION
**Pull request by echarlaix**
This PR adds the export support of models with a multiple-choice classification head, resolving [#16695](https://github.com/huggingface/transformers/issues/16695)

This includes the following additions:
* The `"multiple-choice"` feature was added to the corresponding model topologies
* The dummy inputs are generated to match the expected inputs shape which includes an extra dimension corresponding to the number of candidates answers
* The `inputs` method of the models corresponding `OnnxConfig` were modified to support the additional dynamic axis corresponding to the number of candidates

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Looks good on my side, thanks for working on this! Just make sure to run `make fix-copies` and `make style` on your branch to fix the quality issues.

**Comment by echarlaix** [comment]
Thanks for the  review as well as the reminder @sgugger !

**Review by lewtun — APPROVED** [review]
Thanks for adding this @echarlaix - it looks great to me!

I think there's a few more models that support multiple choice classification and have an ONNX export:

* BigBird
* Data2VecText
* Electra
* FlauBERT

Would you like to include them in this PR too?

We should also do a final run of the slow tests before merging to make sure everything passes:

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py
```

File: src/transformers/onnx/config.py [hunk 1]
```diff
@@ -295,6 +309,17 @@ def generate_dummy_inputs(
...
             # Generate dummy inputs according to compute batch and sequence
             dummy_input = [" ".join([preprocessor.unk_token]) * seq_length] * batch_size
+            if self.task == "multiple-choice":
+                # If dynamic axis (-1) we forward with a fixed dimension of 4 candidate answers to avoid optimizations
+                # made by ONNX
+                num_choices = compute_effective_axis_dimension(
+                    num_choices, fixed_dimension=OnnxConfig.default_fixed_num_choices, num_token_to_add=0
+                )
+                dummy_input = dummy_input * num_choices
+                tokenized_input = preprocessor(dummy_input, text_pair=dummy_input)
```
Comment (michaelbenayoun) [inline src/transformers/onnx/config.py:319]: I would just add a small comment to make clear what's happening here.
What do you think?

```suggestion
                # Shape is [batch_size x num_choices, seq_length]
                tokenized_input = preprocessor(dummy_input, text_pair=dummy_input)
```

**Review by michaelbenayoun — APPROVED** [review]
Perfect, thanks a lot for taking care of this @echarlaix !

**Comment by echarlaix** [comment]
Thanks for the reviews @lewtun and @michaelbenayoun. 
I have added some comments to make things clearer as well as the BigBird, Data2VecText, Electra and FlauBERT models support. Also when running the command line `RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py`, all the tests are passing.

**Comment by lewtun** [comment]
Thanks for iterating on this @echarlaix - it looks great!

**Pull request merged**